price-estimator/src/models.rs

```diff
@@ -79,11 +79,25 @@ pub struct MarketsResult {
     pub bids: Vec<TransitiveOrder>,
 }
 
+/// Sort orders by their price and merge orders with the same price into one.
+fn sanitize_orders(mut orders: Vec<TransitiveOrder>) -> Vec<TransitiveOrder> {
+    orders.sort_unstable_by(|a, b| a.price.partial_cmp(&b.price).expect("price isn nan"));
+    let mut result = Vec::<TransitiveOrder>::with_capacity(orders.len());
+    for order in orders {
+        match result.last_mut() {
+            #[allow(clippy::float_cmp)]
+            Some(last) if last.price == order.price => last.volume += order.volume,
```

Comment:
I couldn't find any explicit documentation, but I assume `display_fromstr` is precise enough to give every distinct float a distinct string (as in there are not two floats evaluating to 199.79999999999998). Otherwise we would need to compare the string representations here.

Reply:
`serde` uses https://docs.rs/ryu/1.0.3/ryu/ for float to string conversion, just FYI.

Reply:
I thought about this too but I came to the conclusion that this is what you would expect from number formatting. I didn't check then but through ryu I find:

> Ryu generates the shortest decimal representation of a floating point number that maintains round-trip safety. That is, a correct parser can recover the exact original number.